test/client/domains.test.js

```diff
@@ -0,0 +1,106 @@
+/* @flow */
+import { ENV } from "@paypal/sdk-constants/src";
+import { beforeEach, describe, it, expect } from "vitest";
+
+import { sdkClientTestGlobals } from "../globals";
+import {
+  getPayPalDomainRegex,
+  getVenmoDomainRegex,
+  isPayPalTrustedDomain,
+} from "../../src";
+import { getAuthAPIUrl, getOrderAPIUrl } from "../../src/domains";
+
+beforeEach(() => {
+  window.__ENV__ = "test";
+  window.__PAYPAL_DOMAIN__ = sdkClientTestGlobals.__PAYPAL_DOMAIN__;
+});
+
+describe(`domains test`, () => {
+  it("should successfully match valid paypal domain", () => {
+    const validDomains = [
+      "master.qa.paypal.com",
+      "test-env.qa.paypal.com:3000",
+      "geo.qa.paypal.com",
+      "www.paypal.com:3080",
+      "www.paypal.cn",
+      "www.paypal.cn:3000",
+      "www.mschina.qa.paypal.cn",
+      "www.paypal.com",
+    ];
+
+    for (const domain of validDomains) {
+      expect(domain).toMatch(getPayPalDomainRegex());
+    }
+  });
+
+  it("should not match invalid paypal domains", () => {
+    const invalidDomains = [
+      "www.paypal.com.example.com",
+      "www.paypal.cn.example.com",
+    ];
+
+    for (const domain of invalidDomains) {
+      expect(domain).not.toMatch(getPayPalDomainRegex());
+    }
+  });
+
+  it("should successfully match valid venmo domain", () => {
+    const validDomains = [
```

Comment:
I would expect these would be something to come from a constant, but felt it was a creeping scope to chase that down, too.